### ISSUE TITLE
Tests are not deterministic

### ISSUE THREAD

**Issue by anthonymartin** [popcorndao/workspace#381]
**Observed Behavior** when running tests `yarn hardhat test` on `packages/contracts` the results are non-deterministic. Sometimes the tests pass, and other times, some tests fail.

**Desired state** when running tests  `yarn hardhat test` on `packages/contracts`, the results should be the same after every execution.

**Background** Some tests are using `evm_increaseTime` to speed up time in certain tests and then running assertions after the time has been increased. It's necessary to increase time to simulate certain scenarios such as vaults becoming unlocked, or new state transitions to be allowed.

Ideas to make tests deterministic: 
1. use `evm_setNextBlockTimestamp` instead of `evm_increaseTime` 
2. do not make more than one assertion after calling `evm_increaseTime` (create new tests for  each assertion using `evm_setNextBlockTimestamp`)
3. use `evm_setNextBlockTimestamp` immediately before making any assertion where time influences the expectation

example failed test: https://github.com/popcorndao/workspace/pull/365/checks?check_run_id=3340264596

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 300.0 USDC (300.0 USD @ $1.0/USDC)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/popcorndao/workspace/381/100026322).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/popcorndao/workspace/381/100026322).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $3,622,804.01 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 265 years, 3 months from now.
Please review their action plans below:


**1) [icebert04](https://gitcoin.co/icebert04) has started work.**

Hello @anthonymartin,

I see that you need help when running tests under the yarn hardhat test on packages/contracts.

I have already done 4 bounties on Gitcoin.co
The 1st task was with doing smart contract audit with Alethea AI (originators of the iNFT standard).
I reviewed their ERC-20 code before them sending the contract to the leading auditing DAOs/firms.
Link attached here...
https://gitcoin.co/issue/AletheaAI/Gitcoin-Bounties/1/100026030

Feel free to check on my Github repo on projects that I made on the Blockchain.
Link attached here...
https://github.com/icebert04?tab=repositories

I'll be working on this issue right away.

All the best,
Francis

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/popcorndao/workspace/381/100026322).

**Comment by icebert04** [comment]
@anthonymartin 

I am trying to run your code but I have a few problems...

Due to the `packages/contract` folder both package.json `@popcorn/contracts` and config file `hardhat.config.ts`

The commands `yarn install`, and `yarn hardhat compile` don't seem to work.

Do you know any solutions to this?

Waiting for your response on this... 

Thanks!

**Comment by anthonymartin** [comment]
are you on develop branch? try running `yarn` from the root project
directory

**Comment by icebert04** [comment]
@anthonymartin, Ok, I'll be working on it...

**Comment by icebert04** [comment]
@anthonymartin 

I've run `yarn install` in the develop branch, but the error still says `@popcorn\contracts: Command failed`.

I'm using the readme file for instructions, is there something that I'm missing?

**Comment by anthonymartin** [comment]
Is there more information? what system are you running?

**Comment by icebert04** [comment]
@anthonymartin 
The pc that I'm using is a Windows 10 Pro, and the source-code editor I use is VScode

**Comment by anthonymartin** [comment]
And debug information? feel free to join our discord here to chat about it: https://discord.gg/GRcZGmzp

**Comment by icebert04** [comment]
Ok, I'll join, but please give me some time to review the code..

**Comment by icebert04** [comment]
@anthonymartin I've joined the discord and sent the error I encountered.

Waiting on a response...

**Comment by icebert04** [comment]
Hello @anthonymartin 

Are you online?

Need to ask a few questions...

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 265 years, 2 months from now.
Please review their action plans below:


**1) [rodrigoherera](https://gitcoin.co/rodrigoherera) has started work.**

Hi anthony, how are you?, i could solve your problem!

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/popcorndao/workspace/381/100026322).

**Comment by rodrigoherera** [comment]
Perfect, i've installed the project, the guy's above error was 'cause the install needs to have the .env file.
When i have some news, ill reach you.

**Comment by rodrigoherera** [comment]
@anthonymartin can you send me again the discord invitation?, thnks!

**Comment by anthonymartin** [comment]
https://discord.gg/39P2dC4d

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 265 years, 2 months from now.
Please review their action plans below:


**1) [antonyip](https://gitcoin.co/antonyip) has started work.**

make tests deterministic by using setNextBlockTime instead of evm_increasetime.
check assertions.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/popcorndao/workspace/381/100026322).

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 265 years, 2 months from now.
Please review their action plans below:


**1) [mecadisruptor](https://gitcoin.co/mecadisruptor) has started work.**

Do a tear-down and full review of tests, look for inadvertent race conditions. Try out ideas mentioned in description

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/popcorndao/workspace/381/100026322).

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__The funding of 300.0 USDC  (300.00 USD @ $1.0/USDC) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $4,458,230.15 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by anthonymartin** [comment]
this is mostly fixed by https://github.com/popcorndao/workspace/pull/442
